src/panel-layout/internal.tsx

```diff
@@ -0,0 +1,153 @@
+import { getBaseProps } from '../internal/base-component';
+import PanelResizeHandle from '../internal/components/panel-resize-handle';
+import { fireNonCancelableEvent } from '../internal/events';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { useControllable } from '../internal/hooks/use-controllable';
+import { SomeRequired } from '../internal/types';
+import useContainerWidth from '../internal/utils/use-container-width';
+import { PanelLayoutProps } from './interfaces';
+
+import styles from './styles.css.js';
+import testStyles from './test-classes/styles.css.js';
+
+const DEFAULT_PANEL_SIZE = 200;
+
+type InternalPanelLayoutProps = SomeRequired<PanelLayoutProps, 'panelPosition' | 'resizable' | 'display'> &
+  InternalBaseComponentProps;
+
+const InternalPanelLayout = React.forwardRef<PanelLayoutProps.Ref, InternalPanelLayoutProps>(
+  (
+    {
+      panelPosition,
+      panelContent,
+      mainContent,
+      defaultPanelSize,
+      panelSize: controlledPanelSize,
+      resizable,
+      onPanelResize,
+      minPanelSize,
+      maxPanelSize,
+      i18nStrings,
+      display,
+      panelFocusable,
+      mainFocusable,
+      __internalRootRef,
+      ...props
+    },
+    ref
+  ) => {
+    const baseProps = getBaseProps(props);
+
+    const resizeHandleRef = React.useRef<HTMLDivElement>(null);
+    const panelRef = React.useRef<HTMLDivElement>(null);
+    const [containerWidth, rootRef] = useContainerWidth();
+
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        focusResizeHandle() {
+          resizeHandleRef.current?.focus();
+        },
+      }),
+      []
+    );
+
+    const [panelSize = DEFAULT_PANEL_SIZE, setPanelSize] = useControllable(
+      controlledPanelSize,
+      onPanelResize,
+      defaultPanelSize ?? minPanelSize,
+      {
+        componentName: 'PanelLayout',
+        controlledProp: 'panelSize',
+        changeHandler: 'onPanelResize',
+      }
+    );
+
+    const actualMaxSize = Math.min(maxPanelSize ?? containerWidth, containerWidth);
+    const actualMinSize = minPanelSize ?? 0;
+    const actualPanelSize = Math.max(Math.min(panelSize, actualMaxSize), actualMinSize);
+
+    const resizeHandlePosition = panelPosition === 'side-end' ? 'side' : panelPosition;
+    const resizeProps = useResize({
+      currentWidth: actualPanelSize,
+      minWidth: actualMinSize,
+      maxWidth: actualMaxSize,
+      panelRef: panelRef,
+      handleRef: resizeHandleRef,
+      position: resizeHandlePosition,
+      onResize: size => {
+        setPanelSize(size);
+        fireNonCancelableEvent(onPanelResize, { totalSize: containerWidth, panelSize: size });
+      },
+    });
+
+    const mergedRef = useMergeRefs(rootRef, __internalRootRef, ref);
+
+    const wrappedPanelContent = (
+      <div
+        className={clsx(styles['panel-content'], display !== 'main-only' && testStyles.panel)}
```

Comment:
Why do we choose to hide but render the panels depending on the display prop instead of skip them from rendering? Should we then remove the ARIA props and tab index, too?